File: react/car_rental_react/src/components/Customer.js
import { useState } from "react";
import CustomerService from "../services/CustomerService";

const Customer = () => {
    const [cars, setCars] = useState([]);
    const customerService = CustomerService();

    const show = async () => {
        try {
            const response = await customerService.adminDashBoard();
            setCars(response.data); // ✅ use response.data if Axios
        } catch (error) {
            console.error("Error fetching customers", error);
        }
    };

    return (
        <div>
            <table border="3" align="center">
                <thead>
                    <tr>
                        <th>Customer Id</th>
                        <th>First Name</th>
                        <th>Last Name</th>
                        <th>Email</th>
                        <th>Phone Number</th>
                    </tr>
                </thead>
                <tbody>
                    {cars.map((item) => (
                        <tr key={item.customerID}>
                            <td>{item.customerID}</td>
                            <td>{item.firstName}</td>
                            <td>{item.lastName}</td>
                            <td>{item.email}</td>
                            <td>{item.phoneNumber}</td>


                        </tr>
                    ))}
                </tbody>
            </table>
            <input type="button" value="Show Customers" onClick={show} /> <br /><br />
        </div>
    );
};

export default Customer;

File: Asp.NetCore/Aug_25_Car_Rental_Core/Aug_25_Car_Rental_Core/Models/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Aug_25_Car_Rental_Core.Models
{
    public class Customer
    {
        [Key]
        [Column("customerId")]
        public int CustomerID { get; set; }

        [Column("firstName")]
        public string? FirstName { get; set; }

        [Column("lastName")]
        public string? LastName { get; set; }

        [Column("email")]
        public string? Email { get; set; }

        [Column("phoneNumber")]
        public string? PhoneNumber { get; set; }

    }
}
